components/constellation/constellation.rs

```diff
@@ -2367,6 +2367,17 @@ impl<Message, LTF, STF> Constellation<Message, LTF, STF>
         ReadyToSave::Ready
     }
 
+    fn clear_pending_loads_for_frame(&mut self, frame_id: FrameId) {
+        let pending_pipelines = self.pending_frames.iter()
+            .filter(|frame_change| frame_change.frame_id == frame_id)
+            .map(|frame_change| frame_change.new_pipeline_id)
+            .collect::<Vec<_>>();
```

Comment:
The `collect` is to please the borrow checker. The iter immutably borrows `self` and closing a pipeline mutably borrows `self`.